Feat/planner

ANaq21 opened the pull request:
Planner UI mostly fleshed out and integrated. Few feature implementation/bugs remain

- Flashing when Hovering over a class
- Populate the onboarding series drop-down with relevant regress (when adding a new degree)
- Review how Evaluator Data is handled on frontend based on any changes Oscar continues to make

ANaq21 committed:
Onboarding series of pages in progress 9/19

ANaq21 committed:
Validation + Initial Planner w/ lots of bugs

ANaq21 committed:
commented out year

ANaq21 committed:
Drag and Drop finished

ANaq21 committed:
fixed build failure

ANaq21 committed:
in progress requirments section

ANaq21 committed:
Added Add Sem and Year, but error scrolling broken

ANaq21 committed:
fileUpload Firebase changes

ANaq21 committed:
initial hovering feature

ANaq21 committed:
fixed semesters overlapping on resize

ANaq21 committed:
drag and drop changes

ANaq21 committed:
Edit Plans Feature

ANaq21 committed:
Dropdown Menu Changes

joalen commented:
1. Hmm...I feel like the "stickiness" doesn't retain for the program spec pipeline and it shows; just minor things, but would be great if you can patch it up real quick. 
2. Same issue as 1. also happens whenever someone interacts with the active/inactive dropdown 


WOW! I'm really impressed with the evaluator and the degree planning page. Kudos to you both @ANaq21 and @sudocanttype! Y'all really pulled together and this sets the baseline for how the degree planner works

joalen committed:
Merge branch 'develop' into feat/planner

joalen committed:
Update Onboarding.tsx

- Fixed issue with outside click for onboarding event
- // TODO: add conditional logging for just debug envs only

joalen committed:
Removed vertical stacking in file upload

- for debugging purposes, this is great, but I think it adds too much visual strain for the user. We can keep it simple like this to show that a file uploaded and parsed.

joalen committed:
Update planner page

- got planner page to respect the Figma designs, which looks amazing!
- removed hardcoded endpoint
- lock global scroll for page and respect dark navbar for onboarding and program spec process
 - // TODO: Fix a few dropdowns and get their interactions working

joalen committed:
Fix dropdown interactability

- the dropdowns weren't interactive; therefore, fixed by adding in the propagation functionalities
- added z-index + padding to help the highlighting feel not less awkward
 - // TODO: add implementation from either CRUD or some other place to be able to fetch available programs for "graduate" and "undergraduate" when populating dropdowns in ProgramValidationB

joalen committed:
Added disclaimer text to onboarding stage + micro-fixes with container

- container needs to naturally scroll; therefore, added that with the option of hiding the scrollbar
- added the disclaimer text as done in the Figma

joalen committed:
Dynamic populate dropdowns + remove option for planner + custom load

- for the dropdowns, it was set to use mock dropdown data, but now that's changed by fetching ONCE from CRUD from a degree catalog master dump
- remove option made so that users can remove semesters (and even years as needed...only applies for a year that contains a single semester)
- the loading page now contains a SAGE spinning icon with the green loading to match with our theme

joalen committed:
Ability to re-upload transcript

- needed to make conditional rendering of the small-text to allow switching between "uploading" transcript vs. "manual" editing of programs
- // TODO: now, the next thing is adding that disclaimer text for the sidebar

joalen committed:
Vertical Collapse + Disclaimer Text for Planner

- added disclaimer text for planner page as seen in the Figma
- also added the vertical collapse

joalen committed:
Add Test and Transfer Credits + bouncy arrow

- added the test and transfer credits containers to show that the transcript parsed out even those sections too and allow users to really confirm what all things they have
- additionally...removed the scrollbar and instead replaced with a bouncy arrow to show scrolling further as needed

joalen committed:
Made program spec process smooth + back button + changed active/inactive

- got some feedback and the transcript upload section was unnecessarily complicated
 - therefore, when upload happens, we immediately jump to the next screen unless an error comes, which prompts the user and prevents advancement
- back button is of course added, but there in case users wanted to adjust settings
- removed some log statements to prevent console flooding
- in addition, rather than "active" and "in-active," we delineate between "in progress", "completed", "withdrawn"
 - these options are good as it gives more info to the user and allows them to see at hand what they're dealing with in the planner program spec pipeline

joalen committed:
Better scroll for UX in Class Validation

- ok, so I felt that the arrow "bouncy" effect wasn't that great and to account for average person, the scroll bar is back but in theme with the actual SAGE branding + a bit further from the container to not make it as awkward.
 - // TODO: implement for FIrefox and other browsers using this

joalen committed:
course fix deduplication + removal (#96)

* Prevent dedupe of already added courses onto a planner

- added logic to prevent adding an already added course to a term or to another term with an error message
- just uncovered a bug where if you add all three semesters, remove just the one semester...can be middle or first, and then add a semester, it somehow jumps to the next semester??
 - this will be filed as another issue for me to keep track of, but a really good catch

* Removal logic of courses
added removal logic for courses to prevent courses from getting "stuck" onto a semester

joalen committed:
Fixed bug for adding/removing semesters

- like #97, it prevents users from accidentally having a semester for another year populate in a particular year.
 - instead, we find the missing one and auto populate as needed

bnguyen1212 committed:
fix: Firefox scrollbar styling compability issue
- simple inline styling fix from CSS Scrollbars Styling Module Level 1 for non-webkit browsers
- existing webkit behavior preserved
- similar green styling but limited gradient and width control

joalen committed:
Merge pull request #101 from acmutd/fix/cross-browser-scrollbar

fix: Firefox scrollbar styling compability issue

joalen committed:
Planner Sidebar Uniformity to SAGE

- made the planner sidebar more uniform with the main Chatbot interface
 - therefore, added in the white box and moved the disclaimer text below the container
 - made the button a bit bigger for accessibility (it's really just trying to be uniform with the Chatbot interface)
- moved out the logic for isExpanded to be invoked by the Planner interface rather than all baked into Sidebar
 - changed some icons to match SAGE's Chatbot interface

joalen committed:
Add tutorial screen for Planner (#103)

* Initial Joyride Tutorial System

- added in just a few UI tutorial steps but the next commit will introduce more interaction with the user so that they know how to use more of the interface and ways around it.
 - I'm hoping that by using this, we can expand more on this tutorial when adding new feature + improve UX

* Added more description to steps

- I was originally going to add more steps but found that difficult due to how React's state is hard to intercept with the ever evolving Virtual DOM you'd need to work with.
 - therefore, users can take special care in reading the tutorial screening and it should benefit them

* Removed unneeded comment

- forgot to erase that comment

joalen committed:
Made locking/unlocking semester functional

- apparently the lock/unlock button did nothing, so now we added some usability to that, which should now do the following:
 1. when a semester is locked, the drag-and-drop is blocked, meaning you can't add classes into that semester
 2. If a semester contains a course and is locked, that course cannot be moved back to enforce a locked semester as needed
 3. Unlock will make it functional again

- def for the "clear courses" function, there should be a check to see if the semester is unlocked for it to clear...me thinks that if locked, the operation is not allowed and can be made as a warning (like the error modal `setError()` I believe does the trick that tells the user that they'd need to unlock the semester to clear all courses

joalen committed:
Tutorial button addition for Planner + add more spacing (#106)

- the collapsed version of the navbar did not align with the Chatbot's navbar, so this is fixed in terms of the rounding
 - additionally...made sure to "scoot" the navbar a little close to the left boundary so that more space can be made for the planner page (and make it open)
- like Evan mentioned, I think adding a tutorial button for both places would be nice and can allow users to go through it again if needed

sudocanttype committed:
added logic to handle calculating completion status from the frontend + text to make the tracks clear

sudocanttype committed:
reworked the hover mechanism so that it now works

sudocanttype committed:
modified the hover of planned courses to show that they are planned rather than default

sudocanttype committed:
placed variable for coursebox

sudocanttype committed:
sidebar changes for the new indicator

bnguyen1212 committed:
feat: clear semester courses (#110)

SemesterBox.tsx:
- check locked constraint in SemesterBox, pass setError() from parent to show error msg
- added cool jsx

Planner.tsx:
- implemented handleClearSemester logic w/ deep copy
- reused existing delete modal with additional 'action' property
- abused conditional rendering
- simplified error scroll logic

sudocanttype committed:
Feat/fix course hover (#109)

* added logic to handle calculating completion status from the frontend + text to make the tracks clear

* reworked the hover mechanism so that it now works

* modified the hover of planned courses to show that they are planned rather than default

sudocanttype committed:
added logic to handle calculating completion status from the frontend + text to make the tracks clear (#108)

sudocanttype committed:
added logic for the planner to handle returning courses back to suggested state after semester/year deletion

sudocanttype committed:
merge weeee

joalen committed:
Merge branch 'develop' into feat/planner

joalen committed:
Bug: location added twice

- whoops, merge conflicts

joalen committed:
Bug: Dev Banner hid onboarding navbar

- the navbar in the main onboarding page prevented the navbar from showing so this should work
 - applies to also Chatbot's Navbar as well (which looks to be borrowed from profile)
- adjusted the container size for checking if classes are correct due to it looking a little too large in terms of height
- there was a bug in which if u went from the onboarding screen of planner to say, chatbot, it'd render the navbar in that darkmode still...therefore, added in the mutationobserver to the ChatbotNavbar to allow it to catch that in the meanwhile
 - additionally, unmount the component so that the onboarding attribute isn't applied to everywhere

bnguyen1212 committed:
Fix/planner data refresh (#112)

* fix: re-upload properly triggers re-onboarding
- added necessary props to Onboarding.tsx
- added trigger chain from Sidebar->Planner->PlannerPage
- simplified redudant states and handlers
- Instead of having Sidebar mount individual components, have UI trigger send re-onboard state all the way up to PlannerPage and go through flow again
- added necessary handlers to PlannerPage to restart onboarding at "Programs" modal step

* fix: dropdown portal clicks no longer close Onboarding modal
- Click-outside handler detects Radix portal of SelectContent child, which renders outside DOM
- File-upload no longer removes items to preserve localStorage state: setItem already overwrites if key is the same, no need to removeItem and then setItem, causes transcriptData cache to disappear in certain flows

* fix: manual edit flow repopulates planner page
- map updated programs from ProgramValidationA to transcriptData format
- pass back updated transcriptData from ClassValidationA
- added transcriptData as dependency to transformedSemesters to recalculate when transcriptData changes
- cleaned up floating props and handlers

* Bug: add in manual revision of programs

- tiny fix for the sake of completion, which will automatically repopulate that "Or click here to fill manually"
 - this came because we needed to carry in the FileUploader's other props which is fine

---------

Co-authored-by: Alen Jo <jo.alen1@outlook.com>

sudocanttype committed:
merge

sudocanttype committed:
merge

joalen committed:
Add "unset" course planned for clear semesters

- added in the same "unset" state filter for also clearing a semester (remove it works, but clear too needs it)

joalen committed:
Merge pull request #111 from acmutd/feat/#195_course_planned_indicator

Feat/#195 course planned indicator

joalen committed:
Add loading animation to upload screen

- added in a subtle loading animation that shows the user that their transcript is processing

joalen committed:
Changed loading text to black + bug in "manual option" after back btn press

- the loading text was originally the SAGE green color, but it was hard to read, so we made it black (for accessibility purposes)
- When I uploaded my transcript and pressed the back button, there was a "race condition" in which you'd have transcript data appearing as not null for initial onboarding steps...now I removed it to just being not a first time onboardee
 - this move was made so that across different browsers, this functionality works BUT I will change out the isFirstTime functionality in which we retrieve from the backend if non-existent on localStorage

joalen committed:
Merge branch 'develop' into feat/planner

joalen committed:
Improve vertical stacking of courses

- improves UX capability by creating a course pagination carousel rather than rote listing all courses and having users scroll down as much

joalen committed:
Add tokenization to Evaluator (#116)

- whoops, needed to add in the token for the plan evaluator

sudocanttype committed:
test

joalen committed:
Styling + Expand categories as needed

- wanted to align most of the categories like the Figma since that feels more professional and not as "airy" I think...the pill design was too old me thinks
- categories now only expand for if things are not complete or not started; the aim of that is to make students see what courses they need to do
- above all, an overhaul of needed styling changes since it felt clunky

sudocanttype committed:
Revert "test"

This reverts commit fbfe2eb3dc6b5b845a400eb0a3fc1a15003b61b1.

joalen committed:
Better rounding style

- figured that the pill-like portion of the courses felt "bouncy" and "airy"; improved by making it slightly pill but a tiny bit boxy

joalen committed:
Styling of semester boxes

- better styling of the semester boxes to be like the Figma
- added in a grayish backdrop in the drop zone to indicate its presence

joalen committed:
Merge pull request #117 from acmutd/fix-professional-planner-redesign

Redesigned planner to be a bit more polished

joalen committed:
Bug: Dropdown not in front of navbar for planner

- turns out the z-index was wrong; therefore, adjusted so that it's a little higher than the navbar to make it appear like it's in front

joalen committed:
Tablet Mode - Planner

- made it auto-collapse the sidebar to show maximum viewing area for the planner area
- in addition, added touch support within the planner to drag-drop courses
 - figured we can expand support to utd students who may use iPads or tablets to do their planning

joalen committed:
Merge remote-tracking branch 'origin/develop' into feat/planner

joalen committed:
Phone version of Planner

- looked at Mercedes design and I think we can implement it
 - the only changes needed to be made are to rather than drag-drop-courses, is to now add and delete courses  (//TODO: implement add/delete logic for a coursebox)
- other than that, the phone version of planner is viewable and the sidebar collapses JUST like Chatbot. Same might be done for erm profile page, but I'll get to that once Planner is fully complete

joalen committed:
Phone view: Removing a course

- remove functionality to help with getting rid of a course via phone view
- added in the onAdd handler in preparation for adding a course from the sidebar to a chosen semester as required for better planning

joalen committed:
Phone view: Adding a course

- ok, I got the add course logic fixed and working for phone view
 - checked for all possible edge cases except for if a user decides to have two phones and tries to do a "race condition" --> that gets blocked by erm the save state plan that Bao works on; no problem there

joalen committed:
Phone view: Course Tooltip Drawer

- in phone view, I found it kinda nice to make it a drawer that opens from the bottom to show a student on their phone for what that course is and of course the status of the course
- there's also additions made for tablet users where if you don't have a cursor (usually that means you cannot hover), then you have the privilege of tapping that course box while being able to drag the course
- in phone mode, you can press the icon and it'll show you the bottom drawer without fail
- desktop has the hovering capability AND I guess iPads with mouse keyboards (need to actually procure one OR get macOS Simulator to do this for me...testing needed but can be achieved)
- other stuff as well, but all that this commit does is to serve phone and tablet users wanting to use planner without having to sit at their desktop...I'd love to cater to a wide audience if I can :)

joalen committed:
Add additional fixes to resizing on phone mode and tablet mode

- issue in where the pointer attribute caused the checkHover to fail, so removed that
 - mainly just doing boolean logic in figuring out how to get events fired in the HTML and render a proper experience
- added in a condition to help re-expand the sidebar back if going to desktop mode
- considered UX for when ppl want to put SAGE as a smaller window for their desktop needs
 - considered just phone and tablet, but also remembered that ppl can resize windows for planning (esp. for enrollement appts where it may not be desired to have a full screen and then switch to another tab)
- made sure to remove "dead space" found near the disclaimer text so that the sidebar can increase height for more categories to be seen
 - this is especially good for when things roll out for trying out other majors/minors/etc if desire to switch and potentially other fun features up ahead

joalen committed:
Bug: Missing Navbar in Onboarding Stage

- forgot that for app.tsx, I need to add back in the Navbar if we are not in the onboarding stage, so now this should be good to go!

joalen committed:
Tooltip modal for Transcript Retrieval

- most users won't know what an unofficial transcript is as a safe assumption; therefore, added in a tooltip to walk through steps on how to retrieve an unofficial transcript and have it saved

joalen committed:
Fixed actions with remove and add cancel btn

- bug within testing of how the remove button for a new program entry did nothing and while this isn't terrible, it can confuse the user
 - therefore, when a new program is made, a cancel button will show up instead
- additionally, added clarification in UX by adding in a cancel button between the save and remove buttons to help the user not feel as if they committed a permanent action

joalen committed:
Re-style class validation page + icon coloring + window resizing for program validation

- so i noticed this in testing but the program validation modals could use a bit of mobile resizing
- class validation page has restyling done to be aligned to brand standards from Figma
 - icons are colored to distinguish and "make it lively"

joalen committed:
Better resizing of onboarding flow + file upload modals

- tested using an iPhone SE and it felt cramped; therefore, added in a few bit of adjustments to the text, modal sizes, and the drop zone.
- also, now the onboarding flow is responsive across a phone, tablet, laptop, or desktop mode

sudocanttype committed:
implement in-coursebox notices for coreqs

sudocanttype committed:
added warning for users when a class with coreqs are planned but the coreqs have not been added

sudocanttype committed:
remove horizontal line on the bottom when the coreq section is there

sudocanttype committed:
added location of the coreqs for easier access

sudocanttype committed:
show only the classes that are available

sudocanttype committed:
make the changes localized to a single semester

joalen committed:
Merge pull request #123 from acmutd/feat/210_coreq_frontend

Feat/210 coreq frontend

bnguyen1212 committed:
- refactored to create compound PlannerState with Sidebar and Planner data
- added UI for save button and unload warning
- added isLocked as part of semester state
- saving works locally on frontend

joalen committed:
Warning system for SemesterBox

- rather than keeping it static to just co-requisites, it's better if we made an interface that supports multiple-types of warnings
 - in the near future, we may have to accomodate certain warnings to help students see immediately what's wrong
- multiple warnings now collapse into a notification count droplet that allows to see multiple notifications at the same time
- moving some types and utils here and there -- don't like code getting messy (although it's ok for now) :(
- made sure React portal did not interfere with desktop as that doesn't play nicely
 - both mobile and desktop are separated from that instance and should be good to go
- added in logic that helps us detect rules on credit hour handling for undergrad/grad distinctions AND depending on what catalog version they have, then the rules changed, so this adapts for that
 - all of them have friendly messages to help students out and being aware of their plan

bnguyen1212 committed:
- updated plannerData to include multiple plannerStates for multi-versioning

bnguyen1212 committed:
- added UI elements for planner version interactions
- implemented plan management functions

joalen committed:
Add Gap Semester

- added in gap semester support so that we prevent creation of a semester box for an already passed semester

joalen committed:
Make sidebar clickable + add symbol to show clickable

- like the chatbot page, planner needs to follow the same branding; therefore, added in the symbol to show clickable sidebar to expand

joalen committed:
Switch Icons for Sidebar Toggle

- these icons were backwards; therefore, switched icons to make it align with the sidebar in the chatbot page for uniformity

joalen committed:
Polished Year Section of Degree Planner

- from the Figma branding, we will now use a YearDivider with the hamburger menu to indicate the many options to manipulate the "year" part of a degree plan
 - rather than just keeping a single Add Year button, now, it's housed in a dropdown modal within the hamburger
   - this allows for things like bulk clearing all user-created semesters and removing an entire year (without relying on deleting all the semesters beforehand)
- scrollbar was annoying in the degree planner side, so removed it from view
- also, prevent double-clicking for the expand sidebar button + being able to click anywhere in the sidebar to expand

joalen committed:
Replace joyride with driver.js

- found that Joyride is kinda finicky with the interactive-ness that our tutorial brings...therefore...decided to give up joyride and use driver.js for tutorial system
 - driver.js is becoming now the de-facto onboarding system either way and has more support
- I'll ofc change the buttons to be in theme of SAGE, but for now, this commit just setups the new system

joalen committed:
Semester Tutorial Added + Tutorial Recall Step

- added in the tutorial recall step so that users know that they can replay the tutorial at any time
- semester tutorial added to the best of my ability
- other steps are inferential and users should pick it up rather quickly

joalen committed:
Switch from Chatbot to Landing Page

- now that planner will be coming...we should let users pick where they want to go rather than navigating them to Chatbot

bnguyen1212 committed:
Merge remote-tracking branch 'origin/feat/planner' into feat/save-planner

joalen committed:
Year and Semester Collapse/Expand

- from some feedback received, we could make the year and semester collapsable/expandable
 - this was done so that you can see the save/load plan state bar above
 - it also helps for planning and preventing having to scroll all the way down for a particular plan

joalen committed:
UI sessionStore

- for planning page, I figured that if a user experiences an intermittent connection or needs to refresh the page, then sessionStore should be able to retain "some" of the UI stuff they did (like expanding years, semesters, sidebar collapse)
  - transient UI states are not added as that would be rather cumbersome to sessionStore
  - also...minor things like category expansions and whatnot aren't included since I hope it wouldn't annoy the user; if need be, we can add it, but not worth it right now

joalen committed:
Redundant Warning Icon in Sidebar Course

- too much UX is happening in that CourseBox; therefore, rather than showing the warning icon, we can remove it and the user will still see through the tooltip that there needs to pair a co-req with it.

joalen committed:
Replaced JSON deep copies with structural sharing

- so this might just be in the worst case...but we can imagine that a student may want to plan a lot of courses and a lot of semesters
 - the original code had it to where we kept on deep-copying the JSON throughout, which could cause potential bottlenecks in serialization/deserilization
  - ofc, there will be limits in place for how many years can be added as well since you can't stay forever (would need to come up with the logistics of how bachelors/masters/doctoral programs work for duration of time)
- Either way...by doing this, we can expect a performance increase in planner page (or so I think)

joalen committed:
Warning tooltip clipping into sidebar fix + refinement of text

- to not make it sound programmatic...language changed to be personal to the user (so user -> your)
- tooltip also for warnings of a semester-box, was clipping inside the sidebar, so that's fixed by pushing it in a React Portal

joalen committed:
Added two additional steps to tutorial overlay for planner

- two steps were added to focus on the collapse/expand functionality of years and semesters so that users know what they are

joalen committed:
Added font-styling for tutorial popover

- got feedback that we should keep consistent font styling for the tutorial overlay

joalen committed:
Clipping of Text for Progress and Actual Header Text

- seems like when trying out other majors, the progress text collides with the actual header text, which could cause points of confusion.
 - fixed by changing some properties to make it wrap instead
 - also, in the offc-hance that overflow happens, I added safety to that as well

joalen committed:
Dropdown Program Truncation

- turns out, with very long program names, the dropdown overflows
 - Radix UI was weird in how it replaced some of the logic I was going to implement for truncation of text with ellipses.
   - Replaced one of the child-selectors where it incorrectly targeted the chevron rather than the value
- also, since we want to keep dropdown sizes constant and not "adaptive," it's easier on the user's eyes if there's a constant size for the dropdowns

joalen committed:
Misleading text for Sidebar courses

- I think there was an error in preventing the "No courses in this category" in appearing when suggested courses appear.
 - Basically, it was contradictory to have "No courses in this category" AND suggested courses displayed. Therefore, added in a guarded clause that only shows it if we don't have suggested courses to offer
   - This is better for error checking when users do see smth went amiss in our checks for degree evals

joalen committed:
Wrong attribute to category

- auto-complete tricked me here; should be suggested as defined for the category object

joalen committed:
Subcategory creation + windowed dots

- needed to add implementation to split categories into sub-categories for certain cases
- windowed dots due to an edge case that recommended an enormous amount of courses...did not think it was possible, but here we are

joalen committed:
Truncate texts within a Course

- As I was testing, I noticed that the text spills a bit out of bounds of the Course object itself.
 - Figured we could add a truncate to just prevent the text from spilling out

joalen committed:
Prevent Drag-n-Drop if exceeding credits even without approval

- to prevent the edge case of a person adding, say a lot of courses onto a single semester (mainly for quick eval in which we don't want an expensive payload sent), it shows the warning to explain why the drag-n-drop is blocked

sudocanttype committed:
initial stage of an external executable

sudocanttype committed:
oops did not mean to push directly to the save-planner branch

sudocanttype committed:
Revert "initial stage of an external executable"

This reverts commit aafcd1835327c67ec9325ded7375587a35d04d4b.

sudocanttype committed:
implemented normalization for the semesterbox coreq handling

sudocanttype committed:
added normalization to the carousel too

sudocanttype committed:
added implementation of multisem planning

joalen committed:
This is the localStorage clear function

- ok, this is the localStorage clear functionality in which we wipe details on logout if we were to be logged in a public environment
 - no logic is in-place for this but this is good for that reason

sudocanttype committed:
frontend stuff

sudocanttype committed:
prereq utils

sudocanttype committed:
added a collision warning

sudocanttype committed:
prerequisite works

sudocanttype committed:
added more specific locations

sudocanttype committed:
updated warning messages

sudocanttype committed:
update how warnings are shown

sudocanttype committed:
make it so that the warning sign only shows up on prereq warnings

sudocanttype committed:
update warning highlighting

sudocanttype committed:
update warning text

sudocanttype committed:
made the icon for a semester course red

sudocanttype committed:
just added a quick eval button, probably will remove this as we smooth out the UX design

sudocanttype committed:
updated the button + added method to make sure that there is some change to the degree plan first before we suggest more

bnguyen1212 committed:
ui: multi-planner design v2 from lead/director feedback

sudocanttype committed:
only allow planner to be refreshed if the courses that have been planned have changed

joalen committed:
Accessibility Stuff for Planner

- just added some accessibility stuff to help peeps with screen-assistive technology and that; wanna cater to them as well too
 - nothing crazy..just some attributes here and there

sudocanttype committed:
update color scheme so that the warning symbols and the borders are consistent

bnguyen1212 committed:
feat: retrieve degree plan state
- only checks backend once per browser session
- syncs planner with cloud version if newer

bnguyen1212 committed:
Merge remote-tracking branch 'origin/feat/planner' into feat/save-planner

bnguyen1212 committed:
fix: added 5 plan limit to plan manager

bnguyen1212 committed:
fix: use updateTutorialStatus and getProfile to sync tutorial status between sesions
- ty swayam528

joalen committed:
Unused Code in Planner

- caused some Vercel build errors...which was interesting...

joalen committed:
Merge pull request #125 from acmutd/feat/save-planner

Feat/save planner

joalen committed:
Added visual highlighting to current semester/year

- for multi-semester planning, this is a helpful indicator to show students that this semester is where they are right now
 - when collapsing the year containing the current semester, the year will have a subtle "Current" tag on it to showcase that they are in this year

sudocanttype committed:
removed logging and console commnd running

sudocanttype committed:
cleanup main

sudocanttype committed:
reworded button

sudocanttype committed:
ai merged, curious

sudocanttype committed:
made a function to save the degree planner state so that suggesting future classes also saves

sudocanttype committed:
moved the saving functionality to after the evaluation for better loading + snappiness

joalen committed:
Removed unused argument in forEach

- seems like Vercel loves complaining about unused variables...wow

sudocanttype committed:
merged

sudocanttype committed:
merge again

joalen committed:
Merge remote-tracking branch 'origin/feat/planner' into feat/multisem_merged

joalen committed:
Remove repeated stuff from merge issues

- seems like the AI is crap at merging...never trust it again....

sudocanttype committed:
Merge branch 'feat/planner' into feat/multisem_merged

sudocanttype committed:
made the buttons aligned

joalen committed:
Merge branch 'feat/multisem_merged' of https://github.com/acmutd/sage-site into feat/multisem_merged

joalen committed:
Merge branch 'feat/multisem_merged' of https://github.com/acmutd/sage-site into feat/multisem_merged

sudocanttype committed:
Revert "made the buttons aligned"

This reverts commit 875c68f0453c6d04e38efdc29f6a327c68d035d3.

joalen committed:
duplicated lines found and orphaned close brace

- yeah, the above describes it

sudocanttype committed:
made the buttons aligned

joalen committed:
Resized buttons from Save planner

- aligned buttons but also resized them to be like that of the save planner so that styling is consistent across the board

joalen committed:
Merge pull request #126 from acmutd/feat/multisem_merged

Feat/multisem merged

joalen committed:
Storing transcript data to backend

- Ok...added functionality to now let planner save out planner data to localStorage if missing
 - this was needed to help our "quick evaluation" feature work since it apparently needs transcript data to work

joalen committed:
Wrong way to retrieve token

- so the convention here is that Firebase automatically refreshes the token as needed. The cookies thing we need to remove ASAP because that's just stale...rather, we'd need to implement a listener to check and see if a firebase session expires which forces logout

joalen committed:
Replacement of Cookies -> Firebase token

- removed more places that have the retrieval of the firebase token

joalen committed:
Removed storing auth cookies

- Firebase does this automatically and seems like we have the listener setup to auto-logout a user for SAGE

joalen committed:
Make Onboarding Non-Captive

- the current flow of onboarding if a user has already onboarded before feels captive in my opinion
 - therefore, we should style it the same as the modals for the actions of Planner page (like delete/clear).
  - This also alludes to our Figma branding and makes it uniform
- also needed to add a particular <img> attribute to prevent broken Google PFPs from happening

joalen committed:
Scrollable Dropdown -> Search Filtering

- Ok, so I can imagine that a particular student would get frustrated trying to scroll up/down the content for particularly long dropdowns (like program selection)
 - therefore...added in the ability to search + fuzzy match a program at will and incorrect results get filtered of course

joalen committed:
Add explicit typing for parameter `o` + absolute path correction for Combobox + removed unneeded component

- ok...so TypeScript complained of implicit any types; therefore, added in the string type
- additionally...added in the absolute, stickier path to Combobox
- removed an unneeded ProgramComboBox that i created to create inheritance for the combobox, but whatevs

joalen committed:
Coursebox -> Searchbox

- Ok Vercel stop complaining about casing....

joalen committed:
Renamed file for SearchBox

- ughhh...had to disable a flag to get a rename file working because apparently Vercel just wants to be annoying and complain that the file doesn't exist even though it does....cmon now!
 - git being git also didn't track renames of files so I just did that

joalen committed:
Remove duplicate searchbox

- thanks git for not allowing case-sensitive files to go through and thanks Vercel for being rather infuriating in trying to get yourself to deploy right

joalen committed:
Placeholder text now more interactive

- to make SAGE inviting to letting students pick out the minors/certs/etc of interest...we have it to where it cycles and randomizes the programs of interest
 - and I really like this...but the slot machine effect JUST feels so nice mainly due to this being a list of options

joalen committed:
Tutorial steps update

- aligned the tutorial steps to be in-line with the current layout we've set...I think that's all the majors steps needed to be added but more could come and that's ok for the time being right now

sudocanttype committed:
implemented saving of the coreqs + prereqs

joalen committed:
Bug: updated program plans never reflected to evaluator

- turns out, there was a bug in which updated program plans made by the user was never reflected to the evaluator. This is now fixed and good to go!

joalen committed:
Receiver: Profile -> Planner Hot link

- added in the ability to now receive the actual profile to the planner hot link
 - Mercedes had that idea for being able to click on the actual things to point to the planner page BUT I refined it to be able to specifically click on the category of interest in the planner page
   - def a lot of work though to coordinate that, so phew!

joalen committed:
Merge branch 'develop' into feat/planner

joalen committed:
Bug: Profile -> Planner Hotlink did not properly scroll to view and auto-expanded cats

- it was annoying that the categories didn't expand NOR did it actually find the element of interest, so now that should be ok it seems
- additionally...I got the profile -> planner link thingy working.
 - Mobile view has added support for it now

joalen committed:
Racy planner state

- turns out, there was a race condition in which sessionStorage retains the old context after logout, so that's fixed by just clearing it
- there's also a race in which if we logout and then go back to planner, we are dropped into the uploadTranscript page rather than just going to planner itself
  - there's another race I encountered, but lemme check

joalen committed:
Another racy condition of manual localStorage clear

- ok, so I reordered around the initializePlanner functionality to where we FIRST check to see if our planner-state and evaluation are empty...if so, remove that item from sessionStorage and that should make syncPlannerFromCloud rerun
 - afterwards, THEN we want to recheck to see if we actually had an effect from syncPlannerFromCloud(). If still not, then we put them into uploadTranscript page.

sudocanttype committed:
remove excessive JSON parse of the transcript data in quick eval

joalen committed:
Transcript Conflict Resolution System

- Might be the case that a student DOES indeed plan their degree out but then they loose their changes without getting warned of loosing those changes
 - therefore, we present them options to either: "overwrite", "select", or "create a new" degree plan
- still need to fix some kinks in the re-onboard stage or more specifically....why modified classes don't do anything

sudocanttype committed:
updated a field and apparently no one decided to update the frontend

joalen committed:
Load screen hid the actual overwrite modal

- well...that was infuriating...turns out we've never actually had a case in which we prevented the load screen at all. PHEW

joalen committed:
Sidebar Change finally works!

- ok...finally had it to where you can modify the plans even if you are a new user
- there's also sidebar content that changes in reflection to the plan being saved
 - yes..the payload may get bigger BUT very worthy UX experience I believe
- the next issue I found is related to the dropdowns I think for selecting the plans and that prob comes from the filter issue I have with fuse.js

joalen committed:
Merge branch 'feat/planner' of https://github.com/acmutd/sage-site into feat/planner

joalen committed:
Make Planner Organized

- I was so tired just seeing the ui folder spammed with all planner stuff, so that should be moved into the planner folder; must've been an AI thing and we just never touched it but felt to clean it up since it could def make it a headache to maintain

joalen committed:
Resolve more file path issues

- VS code being a not so great IDE, did not alert me to the possible import errors for the ENTIRE project.
 - I might just make a switch to Webstorm or something because VS code just couldn't bother LOL

joalen committed:
Update ProgramValidationA.tsx

- forgot to update this file as well!

joalen committed:
Removed the slice from SearchBox

- ok...so I found it confusing that the searchbox limited to 8 results when we should show ALL the results to the user WHILE allowing them to search and have it filter down.
- for now...I need to figure out the logistics of how to get it to filter results and dropping unrelated ones for when the user is close?
- and maybe even bring back the easy scroll of going up to go fast up or going down to go fast down

joalen committed:
Update SearchBox.tsx

- wow that was simple, literally just change the fuzzing threshold and it works!
 - also, I backtrack on the whole scroll up idea...too much UX overload. Let users search and scroll at their free will; I can imagine that if I hover over something..at the top, it can get finicky and hard to use

joalen committed:
Revert button styling

- yeah no...we are never doing the square like buttons...keep it as rounded pills like we've done before

joalen committed:
Merge branch 'feat/planner' into feat/242_save_prereq_coreq_data

Yeah the usual bring the changes from parent to child type beat

joalen committed:
Merge pull request #130 from acmutd/feat/242_save_prereq_coreq_data

Feat/242 save prereq coreq data

bnguyen1212 committed:
Refactored plan management state
- Created Zustand store with Immer and Persist middleware
- Migrated plan CRUD to centralized store and replaced `plannerData` useState with store selectors

bnguyen1212 committed:
Refactored semester management
- added semester management actions
- used Immer for direct mutation with structural sharing
- migrated state handlers to store actions

sudocanttype committed:
fixed the bugs, some coreq stuff was undone from previous commits (#132)

bnguyen1212 committed:
Refactor unsavedChanges
- directly compares actual state vs saved state
- no need to manually set hasUnsavedChanges
- no save needed in event of change + revert

bnguyen1212 committed:
fix: remove `key={activePlanID}` to fix React unmoun/remount components on active plan switch, causing UI flash

sudocanttype committed:
added a new function to handle calculating course credits + calculate the course credits completed and planned on the fly

sudocanttype committed:
added recursive fn to calculate completion

bnguyen1212 committed:
fix: cloud sync uses store's dedicated sync action to update UI state

sudocanttype committed:
added some new handlers for which sections are opened/closed

sudocanttype committed:
info button now works on mobile

joalen committed:
Merge pull request #138 from acmutd/feat/253_planned_completion_indicators

Feat/253 planned completion indicators

bnguyen1212 committed:
Merge remote-tracking branch 'origin/feat/planner' into refactor/zustand

bnguyen1212 committed:
merge: move getCreditsFromCourseCode to dropCourse action

bnguyen1212 committed:
moved tutorial to separate hook

joalen committed:
Added support for Coursebook Sections

- WOW! Finally got it to work. This only ever works in the desktop mode since mobile is pretty finicky and I just see no point in having that
 - Mobile is designed to be a quick view of the planner without being near a computer
- now if I remember right from Luke, we only ever scrape coursebook info per month, so the data would be out of date. This is fine in my opinion and just means we can save per invocation by running it ONCE and saving that into a resource cache
 - this of course expires at one month lock to reinvoke lambda...hopefully, this can cutdown on costs per se
   - had to work some weird magic using service workers (from my chrome extension days) to make it save into the actual browser cache; why I did this...mainly cause localStorage has limits and i wouldn't want to store even filtered course catalogs there
 - dw, the backend already filters down coursebook sections based on the course of interest

joalen committed:
Long-lived tooltip prevented drag-n-drop operation

- ok added in a fix to get that issue sorted

joalen committed:
Stacking tooltips shouldn't happen when hover

- ok there was lots of tooltips that stacked on each other...we need it back to original behavior so I added in a singleton to enforce that pattern.

joalen committed:
Merge branch 'feat/planner' into refactor/zustand

joalen committed:
Added back coursebookData state thingy

- added back the useState for the Course Book Data cause apparently merge into Zustand did not do crap

joalen committed:
Can't compare functions using the three equals

- apparently, you can't compare functions using the `===` (JavaScript quirks lol). Therefore, I tried removing it BUT it caused the same stacking issue before
 - Now, I just add a ref to track current active tooltip, which still solves the same stacking issue I experienced before while not making eslint complain!

joalen committed:
Merge pull request #139 from acmutd/refactor/zustand

Refactor/zustand
